app/scripts/metamask-controller.js

```diff
@@ -1280,10 +1282,12 @@ module.exports = class MetamaskController extends EventEmitter {
    * @param {*} outStream - The stream to provide public config over.
    */
   setupPublicConfig (outStream) {
+    const configStream = asStream(this.publicConfigStore)
     pump(
-      asStream(this.publicConfigStore),
+      configStream,
       outStream,
       (err) => {
+        configStream.destroy()
```

Comment:
Oh wow, I thought the stream would clean itself up. Great subtle catch.

Reply:
this should be redundant bc `pump` destroys on close
unless this is prevented by some duplex half-open state

Reply:
Ahh yes, you are correct. The reason I thought I needed it was because the `obs-store` stream did not have a destroy method and when I added that, I thought I needed to manually call it.

But this call does not need to be in there, pump will call destroy automatically now that the destroy function is implemented.